lib/sycamore/sycamore/transforms/table_structure/extract.py

```diff
@@ -231,6 +231,56 @@ def extract(
         return super().extract(element, doc_image, union_tokens, apply_thresholds)
 
 
+class HybridTableStructureExtractor(TableStructureExtractor):
+    """A TableStructureExtractor implementation that conditionally uses either Deformable or TATR
+    depending on the size of the table"""
+
+    def __init__(
+        self,
+        deformable_model: str,
+        tatr_model: str = TableTransformerStructureExtractor.DEFAULT_TATR_MODEL,
+        device=None,
+    ):
+        self._deformable = DeformableTableStructureExtractor(deformable_model, device)
+        self._tatr = TableTransformerStructureExtractor(tatr_model, device)
+
+    def _pick_model(
+        self, element: TableElement, doc_image: Image.Image
+    ) -> Union[TableTransformerStructureExtractor, DeformableTableStructureExtractor]:
+        """If the absolute size of the table is > 500 pixels in any dimension, use deformable.
+        Otherwise, use TATR"""
+        if element.bbox is None:
+            return self._tatr
+        width, height = doc_image.size
+        bb = element.bbox.to_absolute(width, height)
+        padding = 10
+        max_dim = max(bb.width, bb.height) + 2 * padding
+        if max_dim > 500:
+            return self._deformable
+        return self._tatr
+
+    def _init_structure_model(self):
```

Comment:
This is not going to be triggered correct?

Reply:
The model server uses it to load models into the actors before any requests come through